src/main/java/List.java

```diff
@@ -0,0 +1,128 @@
+        for (int i = 0; i < numItems; i += 1) {
+            // print index of task
+            System.out.print(" " + (i + 1) + ".");
+
+            // list the details about the task. Based on whether the task is ToDo, Deadline or Event.
+            System.out.println(taskList[i].listTask());
+        }
+    }
+
+    private void printSuccessfulAddMessage(int numItems, Task currTask) {
+        System.out.println(" Got it. I've added this task: ");
+        System.out.println("  " + currTask.listTask());
+        System.out.println(" Now you have " + numItems + " tasks in the list.");
+
+    }
+
+    private void listAddToDo(String sentence, int spaceIndex) {
+        String taskName = sentence.substring(spaceIndex + 1); // taskname rest of string
+        taskList[numItems] = new ToDo(taskName);
+    }
+
+    private void listAddDeadline(String sentence, int spaceIndex) {
+        int byIndex = sentence.indexOf(" /by ");
+        if (byIndex == -1) {
+            System.out.println("Invalid deadline command! Please specify due date.");
+            return;
+        }
+        String taskName = sentence.substring(spaceIndex + 1, byIndex);
+        String dueDate = sentence.substring(byIndex + 5); // rest of string after " /by "
+        taskList[numItems] = new Deadline(taskName, dueDate);
+    }
+
+    private void listAddEvent(String sentence, int spaceIndex) {
+        int fromIndex = sentence.indexOf(" /from ");
+        int toIndex = sentence.indexOf(" /to ");
+        if (fromIndex == -1 || toIndex == -1) {
+            System.out.println("Invalid event command! Please specify start and end time.");
+            return;
+        }
+        String taskName = sentence.substring(spaceIndex + 1, fromIndex);
+        String startTime = sentence.substring(fromIndex + 7, toIndex);
+        String endTime = sentence.substring(toIndex + 5);
+        taskList[numItems] = new Event(taskName, startTime, endTime);
+    }
+
+
+    public void listAdd(String sentence) {
+        int spaceIndex = sentence.indexOf(' ');
+        if (spaceIndex == -1) {
+            System.out.println("Invalid command! Please specify task type.");
+            return;
+        }
+
+        String taskType = sentence.substring(0, spaceIndex); // task type up till first space
+
+
+        if (taskType.equals("todo")) {
+            listAddToDo(sentence, spaceIndex);
+        } else if (taskType.equals("deadline")) {
+            listAddDeadline(sentence, spaceIndex);
+        } else if (taskType.equals("event")) {
+            listAddEvent(sentence, spaceIndex);
+        } else {
+            System.out.println("Invalid task type!");
+            return;
+        }
+        numItems += 1;
+
+        printSuccessfulAddMessage(numItems, taskList[numItems - 1]);
+    }
+
+
+    // check if index represented by string is valid, if valid, return index in integer form.
+    // else return -1.
+    private int parseIndex(String index) {
+        int i;
```

Comment:
Naming of variable 'i' can be confusing to readers. May be better to choose names that describe the variable in more detail.